terraform-modules/aws/istio-networking/main-gateway/main.tf

```diff
@@ -38,5 +39,6 @@ data "template_file" "certificate" {
 }
 
 resource "kubectl_manifest" "certificate" {
-  yaml_body = data.template_file.certificate.rendered
+  count     = ${var.enable_certificate 1 : 0}
+  yaml_body = data.template_file.certificate[0].rendered
```

Comment:
As with managed resources, when count, is present it is important to distinguish the resource itself from the multiple resource instances it creates. Each instance will separately read from its data source with its own variant of the constraint arguments, producing an indexed result. [documentation](https://www.terraform.io/language/data-sources#:~:text=As%20with%20managed%20resources%2C%20when%20count%20or%20for_each%20is%20present%20it%20is%20important%20to%20distinguish%20the%20resource%20itself%20from%20the%20multiple%20resource%20instances%20it%20creates.%20Each%20instance%20will%20separately%20read%20from%20its%20data%20source%20with%20its%20own%20variant%20of%20the%20constraint%20arguments%2C%20producing%20an%20indexed%20result.)